src/components/forms/DateInput.jsx

```diff
@@ -0,0 +1,212 @@
+/**
+ * Private component for DateInput that renders Month/Year select inputs.
+ * @param {Object} props react props
+ * @returns {ReactElement}
+ */
+const YearAndMonthSelector = ({ date, localeUtils, onChange, startYear, endYear }) => {
+  const months = localeUtils.getMonths();
+
+  const years = Array.from({ length: endYear - startYear }, (year, i) => i + startYear);
+
+  const handleChange = (e) => {
+    const { year, month } = e.target.form;
+    onChange(new Date(year.value, month.value));
+  };
+
+  return (
+    <form className="DayPicker-Caption border--bottom">
+      <div className="padding--bottom--half">
+        <select
+          name="month"
+          onChange={handleChange}
+          value={date.getMonth()}
+          className="margin--right--half"
+        >
+          {months.map((month, i) => (
+            <option key={month} value={i}>
+              {month}
+            </option>
+          ))}
+        </select>
+        <select name="year" onChange={handleChange} value={date.getFullYear()}>
+          {years.map((year) => (
+            <option key={year} value={year}>
+              {year}
+            </option>
+          ))}
+        </select>
+      </div>
+    </form>
+  );
+};
+
+/**
+ * Private component for DateInput that renders prev/next arrows.
+ * @param {Object} props react props
+ * @returns {ReactElement}
+ */
+const NavArrows = ({ onPreviousClick, onNextClick }) => (
+  <div className="fdsDateInput-navArrows alignChild--right--center">
+    <Flex noGutters>
+      <FlexItem shrink>
+        <IconButton onClick={() => onPreviousClick()} Icon={ActionsArrowLeftIcon} />
+      </FlexItem>
+      <FlexItem shrink>
+        <IconButton onClick={() => onNextClick()} Icon={ActionsArrowRightIcon} />
+      </FlexItem>
+    </Flex>
+  </div>
+);
+/* eslint-enable react/prop-types */
+
+/**
+ * @param {Object} props react props
+ * @returns {ReactElement}
+ */
+const DateInput = ({ futureYears, pastYears, defaultDate, onDateChange }) => {
+  const [selectedDate, setSelectedDate] = useState(defaultDate || null);
+  const [pickerMonth, setPickerMonth] = useState(new Date());
+  const [inputValue, setInputValue] = useState(
+    moment(defaultDate).format(DATE_FORMAT) || ''
+  );
+
+  const { startYear, endYear } = getYearRange(
+    new Date().getFullYear(),
+    pastYears,
+    futureYears,
+    selectedDate
+  );
+
+  const handleYearMonthChange = (date) => {
+    setPickerMonth(date);
+  };
+
+  const handleDaySelect = (date) => {
+    setSelectedDate(date);
+    setInputValue(moment(date).format(DATE_FORMAT));
+    onDateChange(date);
+  };
+
+  const handleInputChange = ({ target: { value } }) => {
+    setInputValue(value);
+
+    if (isValidUserDate(value)) {
+      const parsedDate = moment(value, DATE_FORMAT);
+      setPickerMonth(parsedDate.toDate()); // update the picker month/year in picker as user types
+      handleDaySelect(parsedDate.toDate());
+    }
+  };
+
+  return (
+    <React.Fragment>
```

Comment:
the nitpickiest of nitpicks: Fragment isn't needed here? Also, I've been trying more and more to use `<></>` instead of `<React.Fragment></React.Fragment>`